### ISSUE TITLE
There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "XInputDotNetPure" ...

### ISSUE THREAD

**Issue by aybe** [speps/XInputDotNet#19]
Hello,

While using the library there's a warning happening that I couldn't get rid off:

```
C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets(1819,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "XInputDotNetPure", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
```

The problem is that whenever one would switch to an x86 or x64 architecture, going back to Unity then back to VS, VS will tell that the project/solution have changed and therefore should be reloaded, i.e. it gets fixed until one does Alt-Tab.

Do you think you could provide an `AnyCPU` assembly which would rely somehow on a combination of `LoadLibrary` and `Environment.Is64BitProcess` to load the right version of `XInputInterface.dll` ?

Thank you :D

**Comment by Eluem** [comment]
I'm having the same issue. Is this likely to cause any actual problems though?

**Comment by speps** [comment]
Sorry I forgot about this issue. I might try the `LoadLibrary` thing, as it's already done inside the XInputInterface.dll anyway.

However, that error might come from the test projects themselves. There shouldn't be any issues due to this anyway.

**Comment by OdysseyHome** [comment]
I found this site to be helpful for fixing the warning for myself:

[http://stackoverflow.com/questions/10113532/how-do-i-fix-the-visual-studio-compile-error-mismatch-between-processor-archit](url)

Basically I rebuilt the solution in VS2015 from a cloned repository, but changed the target builds in the Configuration Manager from Active(AnyCPU) specifically to x86 or x86_64 then replaced the dlls in Unity with the appropriate new files.   

As said by speps and the linked article, its just a harmless warning. Microsoft throws this when it auto-detects that your code is not ANY CPU safe despite you the user say that it is via the Config manager.
